client/ayon_maya/api/lib.py

```diff
@@ -4317,10 +4335,28 @@ def create_rig_animation_instance(
         create_context.create(
             creator_identifier=creator_identifier,
             variant=namespace,
-            pre_create_data={"use_selection": True}
+            pre_create_data={
+                "use_selection": True,
+                "lock_instance": options.get("lock_instance", False)
+            }
         )
 
 
+def is_animation_instance(objectset: str) -> bool:
+    """Check if the given object set is an animation instance.
+
+    Arguments:
+        objectset (str): The name of the object set to check.
+
+    Returns:
+        bool: True if the object set is an animation instance, False otherwise.
+    """
+    creator_identifier = cmds.getAttr(f"{objectset}.creator_identifier")
+    if creator_identifier == "io.openpype.creators.maya.animation":
+        return True
+    return False
```

Comment:
I feel like this shouldn't be in `lib.py` - it's quite specific honestly to `load_reference.py` logic. Likely best moved there. This isn't generic enough to be in lib.py.